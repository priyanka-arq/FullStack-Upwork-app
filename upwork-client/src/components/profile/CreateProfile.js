import React, { Component } from "react";
import AuthService from "../../services/auth.service";
import ProfileForm from "./ProfileForm";
import "react-tagsinput/react-tagsinput.css";
import { newProfile } from "../../utils/newProfile";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";
import * as profileActions from "../../redux/actions/profileActions";

export class CreateProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      profile: { ...this.props.profile },
      userReady: false,
      tags: [],
      currentUser: {},
      redirect: null,
    };
  }

  componentDidMount() {
    const currentUser = AuthService.getCurrentUser();
    this.props.profiles.map((profile) => {
      if (currentUser && profile.user === currentUser.id) {
        this.props.history.push(`/user/${currentUser.id}`);
      }
    });

    if (!currentUser) {
      this.props.history.push("/login");
    }

    this.setState({ currentUser: currentUser, userReady: true });
  }

  onSubmit = (profile) => {
    this.props.createProfile(profile);
    this.props.history.push("/profiles");
  };

  render() {
    return (
      <div>
        {this.state.userReady ? (
          <ProfileForm profile={this.props.profile} onSubmit={this.onSubmit} />
        ) : null}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    profiles: state.profiles,
    profile: newProfile,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    createProfile: (profile) => dispatch(profileActions.createProfile(profile)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateProfile);
