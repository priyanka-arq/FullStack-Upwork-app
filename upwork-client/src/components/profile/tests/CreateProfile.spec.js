import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { CreateProfile } from "../CreateProfile";
//import ProfileForm from "../Profileform";
import profiles from "../../../fixtures/profiles";

let wrapped, createProfile, history, userReady;

beforeEach(() => {
  createProfile = jest.fn();
  history = { push: jest.fn() };
  userReady = true;
  wrapped = shallow(
    <CreateProfile
      profiles={profiles}
      createProfile={createProfile}
      history={history}
    />
  );
});

// afterEach(() => {
//   wrapped.unmount();
// });

test("should render Create Profile Correctly", () => {
  expect(toJSON(wrapped)).toMatchSnapshot();
});

test("should render ProfileForm", () => {
  expect(wrapped.find("ProfileForm").length).toEqual(1);
});

test("should handle onSubmit", () => {
  wrapped.find("ProfileForm").prop("onSubmit")(profiles[1]);
  expect(history.push).toHaveBeenLastCalledWith("/profiles");
  expect(createProfile).toHaveBeenLastCalledWith(profiles[1]);
});
