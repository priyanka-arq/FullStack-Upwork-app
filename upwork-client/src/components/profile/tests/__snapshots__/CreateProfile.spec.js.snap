// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Create Profile Correctly 1`] = `
<div>
  <ProfileForm
    onSubmit={[Function]}
  />
</div>
`;
